### PR TITLE
ignore case for config enums. Fixes #738

### LINKED ISSUE #738: Configs have to be cased like the enums
Currently any configuration option enum needs the toml file to have the same casing as it's variants that is: `newline_style` has to be equal to `Unix` not `unix` or `UNIX`

This is because in utils.rs, `impl_enum_decodable` uses the input string without any changes and matches that to the result of `stringify!` which outputs the string as cased used in code.

### PR DESCRIPTION
**Pull request by svmnotn**
This time the code is a true hack, I do not like that I had to switch it from a match to many if statements, but I did not see a different way of doing so.

### CODE REVIEW

**Comment by marcusklaas** [comment]
This looks like a decent solution to me. @nrc, do you approve of case insensitive configs as well?

File: src/utils.rs [hunk 1]
```diff
@@ -212,12 +213,13 @@ macro_rules! impl_enum_decodable {
...
-                    )*
-                    _ => Err("Bad variant"),
-                }
+                use std::ascii::AsciiExt;
+                $(
+                    if stringify!($x).eq_ignore_ascii_case(s) {
+                        return Ok($e::$x);
+                    }
+                )*
+                Err("Bad variant")
```
Comment (nrc) [inline src/utils.rs]: could use if/else rather than if + return for both of these

Comment (nrc) [inline src/utils.rs]: Oh, sorry, nope, I see what you are doing

**Comment by nrc** [comment]
sgtm!

**Comment by nrc** [comment]
Thank you for the PR!

**Pull request merged**